sources/agent/google/google.go

```diff
@@ -0,0 +1,139 @@
+package google
+
+import (
+	"compress/gzip"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/url"
+
+	"github.com/projectdiscovery/uncover/sources"
+)
+
+const (
+	baseURL = "https://www.googleapis.com/customsearch/v1?key=%s&cx=%s&q=%s&start=%d&num=%d"
+)
+
+type Agent struct{}
+
+func (agent *Agent) Name() string {
+	return "google"
+}
+
+func (agent *Agent) Query(session *sources.Session, query *sources.Query) (chan sources.Result, error) {
+
+	if session.Keys.GoogleKey == "" || session.Keys.GoogleCX == "" {
+		return nil, errors.New("empty google keys")
+	}
+
+	results := make(chan sources.Result)
+
+	go func() {
+		defer close(results)
+
+		numberOfResults := 0
+		pageQuery := 1
+
+		escapedQuery := url.QueryEscape(query.Query)
+
+		var size int
+		if query.Limit > 10 {
+			size = 10
+		} else {
+			size = query.Limit
+		}
+
+		for {
+			googleRequest := &Request{
+				SearchTerms: escapedQuery,
+				Count:       size, // max size is 10
+				StartIndex:  pageQuery,
+			}
+
+			if pageQuery > 10 {
```

Comment:
Could you please clarify why we limit pagination to 10?

Reply:
https://developers.google.com/custom-search/v1/reference/rest/v1/cse/list#request
Because the "start" field in the document indicates that up to 100 queries can be made each time.